certbot/tests/main_test.py

```diff
@@ -1404,6 +1404,46 @@ def test_update_account_no_existing_accounts(self):
                  "user@example.org"])
             self.assertTrue("Could not find an existing account" in x[0])
 
+    @mock.patch('certbot._internal.main._determine_account')
+    @mock.patch('certbot._internal.eff.prepare_subscription')
+    @mock.patch('certbot._internal.main.account')
+    def test_update_account_remove_email(self, mocked_account_module, mock_prepare, mock_det_acc):
+        with mock.patch('certbot._internal.main.client') as mocked_client:
+            # Mock account storage and the account object returned
+            mocked_storage = mock.MagicMock()
+            mocked_account = mock.MagicMock()
+
+            mocked_account_module.AccountFileStorage.return_value = mocked_storage
+            mocked_storage.find_all.return_value = [mocked_account]
+            mock_det_acc.return_value = (mocked_account, "foo")
+
+            # Mock client and mock registration body to verify calls are made
+            cb_client = mock.MagicMock()
+            mock_regr_body = mock.MagicMock()
+
+            mocked_client.Client.return_value = cb_client
+            # mocked_account.regr is overwritten in update, requiring an odd mock setup
+            mocked_account.regr.body = mock_regr_body
+
+            x = self._call_no_clientmock(
```

Comment:
nit: Instead of using this and mocking client here, can't we just use `self._call` and assert whatever we want on the returned mock client object?